src/v/redpanda/application.cc

```diff
@@ -815,6 +821,40 @@ void application::admin_register_raft_routes(ss::http_server& server) {
       });
 }
 
+/*
+ * Parse integer pairs from: ?target={\d,\d}* where each pair represent a
+ * node-id and a shard-id, repsectively.
+ */
+static std::vector<model::broker_shard>
+parse_target_broker_shards(const ss::sstring& param) {
+    std::vector<ss::sstring> parts;
+    boost::split(parts, param, boost::is_any_of(","));
+
+    if (parts.size() % 2 != 0) {
+        throw ss::httpd::bad_param_exception(
```

Comment:
Should we throw or simply return a `400 bad request` I don’t get the throw tho. Seems ok, but also expensive. Just a thought on exceptions potentially slowing down other code paths since they do take a machine-global lock.

Reply:
seems to be how Seastar http works. Maybe there are other ways

Reply:
looks like we can write custom http handler to avoid using exceptions, but this is about as slow path as we get. seems like we have bigger things to worry about

Reply:
Sure thing.